server/service/integration_mdm_test.go

```diff
@@ -1829,7 +1827,442 @@ func (s *integrationMDMTestSuite) TestDiskEncryptionRotation() {
+			{Identifier: "T2.2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: nil},
+		},
+		h3: {
+			{Identifier: "T2.1", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+			{Identifier: "T2.2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: nil},
+		},
+	})
+
+	// apply the pending profiles
+	triggerReconcileProfiles()
+
+	// delete a no team profile
+	noTeamProfs, err := s.ds.ListMDMAppleConfigProfiles(ctx, nil)
+	require.NoError(t, err)
+	var g1ProfID uint
+	for _, p := range noTeamProfs {
+		if p.Identifier == "G1" {
+			g1ProfID = p.ProfileID
+			break
+		}
+	}
+	require.NotZero(t, g1ProfID)
+	var delProfResp deleteMDMAppleConfigProfileResponse
+	s.DoJSON("DELETE", fmt.Sprintf("/api/latest/fleet/mdm/apple/profiles/%d", g1ProfID),
+		deleteMDMAppleConfigProfileRequest{}, http.StatusOK, &delProfResp)
+	s.assertHostConfigProfiles(map[*fleet.Host][]fleet.HostMDMAppleProfile{
+		h2: {
+			{Identifier: "G1", OperationType: fleet.MDMAppleOperationTypeRemove, Status: nil},
+			{Identifier: "G2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+			{Identifier: "G3", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+		},
+		h4: {
+			{Identifier: "G1", OperationType: fleet.MDMAppleOperationTypeRemove, Status: nil},
+			{Identifier: "G2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+			{Identifier: "G3", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+		},
+	})
+
+	// delete a team profile
+	tm2Profs, err := s.ds.ListMDMAppleConfigProfiles(ctx, &tm2.ID)
+	require.NoError(t, err)
+	var tm21ProfID uint
+	for _, p := range tm2Profs {
+		if p.Identifier == "T2.1" {
+			tm21ProfID = p.ProfileID
+			break
+		}
+	}
+	require.NotZero(t, tm21ProfID)
+	s.DoJSON("DELETE", fmt.Sprintf("/api/latest/fleet/mdm/apple/profiles/%d", tm21ProfID),
+		deleteMDMAppleConfigProfileRequest{}, http.StatusOK, &delProfResp)
+	s.assertHostConfigProfiles(map[*fleet.Host][]fleet.HostMDMAppleProfile{
+		h1: {
+			{Identifier: "T2.1", OperationType: fleet.MDMAppleOperationTypeRemove, Status: nil},
+			{Identifier: "T2.2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+		},
+		h3: {
+			{Identifier: "T2.1", OperationType: fleet.MDMAppleOperationTypeRemove, Status: nil},
+			{Identifier: "T2.2", OperationType: fleet.MDMAppleOperationTypeInstall, Status: &fleet.MDMAppleDeliveryApplied},
+		},
+	})
+
+	// apply the pending profiles
+	triggerReconcileProfiles()
+	mysql.ExecAdhocSQL(t, s.ds, func(q sqlx.ExtContext) error {
+		mysql.DumpTable(t, q, "host_mdm_apple_profiles")
+		return nil
+	})
+
+	// bulk-set profiles for no team, with add/delete/edit
+	g2Edited := mobileconfigForTest("G2b", "G2b")
```

Comment:
I think our "profile edited" story is buggy - reusing the same identifier/name results in a new internal (db) profile ID, so the reconciliation will try to both remove and install the profile (install sends the content so it should always install the correct thing, but remove only seems to take the profile identifier, which would be a race with the install command as to whether it removes the old one with that identifier, or the newly installed one). This will be tricky to fix (and part of a separate ticket/pr), I think, I'll bring it up in the ticket.